src/librustdoc/config.rs

```diff
@@ -450,15 +450,16 @@ impl Options {
 
         let (lint_opts, describe_lints, lint_cap) = get_cmd_lint_options(early_dcx, matches);
 
-        let input = PathBuf::from(if describe_lints {
+        let input = if describe_lints {
             "" // dummy, this won't be used
         } else if matches.free.is_empty() {
             dcx.fatal("missing file operand");
         } else if matches.free.len() > 1 {
             dcx.fatal("too many file operands");
         } else {
             &matches.free[0]
```

Comment:
(preexisting, style)

```suggestion
        } else {
            match matches.free.as_slice() {
                [] => dcx.fatal("missing file operand"),
                [input] => input,
                _ => dcx.fatal("too many file operands")
            }
```

Reply:
(preexisting, style) and we could also move the `.to_string()` inside the `if/else/match` via `""`→`String::new()`, (`&matches.free[0]`→)`input`→`input.to_owned()`.

Reply:
Used your new style and even removed the (now unnecessary) allocation.